framework/base/Controller.php

```diff
@@ -238,6 +238,33 @@ public function createAction($id)
         return null;
     }
 
+    /**
+     * Finds action method name by action ID
+     *
+     * @param $id string the action ID
+     * @return string|null action method name
+     * @throws \ReflectionException
+     */
+    private function getMethodNameById($id)
+    {
+        $methodName = 'action' . str_replace(' ', '', ucwords(str_replace('-', ' ', $id)));
+        if (!method_exists($this, $methodName)) {
+            $methods = (new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC);
+            $methodNames = array_map(static function ($item) {
+                return [
+                    'original_name' => $item->name,
+                    'name' => strtolower(str_replace('_', '', $item->name)),
+                ];
+            }, $methods);
+            $matchedMethodNames = array_filter($methodNames, static function ($item) use ($methodName) {
```

Comment:
If you created array `original_name => name` then simpler & faster code could be used e.g. `if(array_key_exists(..)){..}` rather than looping always through all method names.